src/main/kotlin/com/bridgecrew/services/ResultsCacheService.kt

```diff
@@ -21,36 +22,55 @@ class ResultsCacheService(val project: Project) {
         this.selectedCategory = category
     }
 
+    fun updateSelectedSeverities(severityList: List<Severity>) {
+        this.severitiesToFilterBy = severityList
+    }
+
     fun getCheckovResultsByPath(filePath: String): List<BaseCheckovResult> {
         return this.checkovResults.filter {baseCheckovResult ->
             baseCheckovResult.filePath == "/${filePath}"
         }
     }
 
-    fun getCheckovResultsFilteredBySeverityGroupedByPath(severitiesToFilterBy: List<Severity>?): Map<String, List<BaseCheckovResult>> {
+    fun getCheckovResultsFilteredBySeverityGroupedByPath(): Map<String, List<BaseCheckovResult>> {
+        val filteredResults = getFilteredResults(emptyList(), emptyList())
+        return filteredResults.groupBy { it.filePath }
+    }
+
+    fun getFilteredResults(categories: List<Category>?, severities: List<Severity>?): List<BaseCheckovResult> {
         var filteredResults = checkovResults
-        if(severitiesToFilterBy != null) {
-            filteredResults = filteredResults.filter { baseCheckovResult ->
-                severitiesToFilterBy.all { includedSeverity ->
-                    includedSeverity == baseCheckovResult.severity
-                }
+        filteredResults = if(!categories.isNullOrEmpty()) {
```

Comment:
I would move the whole logic of filtering by severity and filtering by category to a different function since it's a bit hard to be read... Something like:
```
fun getFilteredResults(categories: List<Category>?, severities: List<Severity>?):
var filteredResults = checkovResults
filteredResults = filterResultsByCategory(filteredResults, categories, selectedCategory)
```

And move the if-else to there

Reply:
I'm with you, I created two other functions for severity / category filters below.
The issue is that we sometimes have an external update to the state, and sometimes we send the filter directly (as null or list). This causes inconsistent behavior. So the smaller functions are dealing with the "general" case - if you provided a parameter filter by it, if not - filter by the global state. The function you're referring to provides a list if it's absent. I'll see if I can change it